pallets/msa/src/lib.rs

```diff
@@ -886,13 +886,16 @@ pub mod pallet {
 			origin: OriginFor<T>,
 			provider_name: Vec<u8>,
 		) -> DispatchResult {
+			let bounded_name: BoundedVec<u8, T::MaxProviderNameSize> =
+				provider_name.try_into().map_err(|_| Error::<T>::ExceedsMaxProviderNameSize)?;
+
```

Comment:
preliminary check for name length, even if actual extrinsic `create_provider` checks for length, a proposal creator wont know until proposal goes though.